feat: L1 contract etherscan verification

spypsy opened the pull request:
- add option to `aztec deploy-l1-contracts --create-verification-json [path]`
- Will generate a structured JSON that can then be used by l1-contracts/scripts/verify-from-json.sh

Fixes https://linear.app/aztec-labs/issue/TMNT-72/need-to-have-contracts-verified-as-part-of-deployment

just-mitch commented on l1-contracts/scripts/verify-from-json.sh:
```diff
@@ -0,0 +1,208 @@
+}
+
+# Map deployment "name" to FQN "<path>:<ContractName>"
+resolve_fqn() {
```
Does this not handle inbox/outbox? Fine if it doesn't, we should just add a follow-on issue.

spypsy replied:
added 👍

just-mitch reviewed (approved):
Probably also want the Slasher and the EmpireSlashingProposer/TallySlashingProposer (depending on the slasher type provided). I'm happy to merge this now and add that as a separate issue.

AztecBot committed:
feat: L1 contract etherscan verification

- add option to `aztec deploy-l1-contracts --create-verification-json [path]`
- Will generate a structured JSON that can then be used by l1-contracts/scripts/verify-from-json.sh

Fixes https://linear.app/aztec-labs/issue/TMNT-72/need-to-have-contracts-verified-as-part-of-deployment

Co-authored-by: spypsy <spypsy@outlook.com>
